caproto/threading/client.py

```diff
@@ -2160,13 +2164,51 @@ def add_callback(self, func):
         Parameters
         ----------
         func : callable
-            Expected signature: ``func(response)``
+            Expected signature: ``func(response, pv)``.
+            The signature ``func(response)`` is also supported for
+            backward-compatibility but will issue warnings. Support will be
+            removed in a future release of caproto.
 
         Returns
         -------
         token : int
             Integer token that can be passed to :meth:`remove_callback`.
+
+        .. versionchanged:: 0.5.0
+
+           Changed the expected signature of ``func`` to add ``pv``.
         """
+        # Handle func with signature func(respons) for back-compat.
+        if len(inspect.signature(func).parameters) == 1:
```

Comment:
2 tests are failing due to this:

`def callback(command, **kwargs):`
having a signature:
```
In [2]: sig.parameters
Out[2]:
mappingproxy({'command': <Parameter "command">,
              'kwargs': <Parameter "**kwargs">})
```

Might consider using `sig.bind(None)` to see if it takes only one positional arg, or `sig.bind(None, None)` (for example) to see if it has the new signature.

Reply:
(The `f(command, **kwargs)` example lives on a new, second test of the deprecated code path.)